FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Multitask/DemoTasks/MultitaskMQTTExample.c

```diff
@@ -751,21 +826,36 @@ static MQTTStatus_t prvMQTTConnect( MQTTContext_t * pxMQTTContext,
     xConnectInfo.pClientIdentifier = democonfigCLIENT_IDENTIFIER;
     xConnectInfo.clientIdentifierLength = ( uint16_t ) strlen( democonfigCLIENT_IDENTIFIER );
 
-    /* Set MQTT keep-alive period. It is the responsibility of the application to ensure
-     * that the interval between Control Packets being sent does not exceed the Keep Alive value.
-     * In the absence of sending any other Control Packets, the Client MUST send a PINGREQ Packet. */
+    /* Set MQTT keep-alive period. It is the responsibility of the application
+     * to ensure that the interval between Control Packets being sent does not
+     * exceed the Keep Alive value. In the absence of sending any other Control
+     * Packets, the Client MUST send a PINGREQ Packet. */
     xConnectInfo.keepAliveSeconds = mqttexampleKEEP_ALIVE_INTERVAL_SECONDS;
 
-    /* Send MQTT CONNECT packet to broker. LWT is not used in this demo, so it
-     * is passed as NULL. */
+    /* Send MQTT CONNECT packet to broker. MQTT's Last Will and Testament feature
+     * is not used in this demo, so it is passed as NULL. */
     xResult = MQTT_Connect( pxMQTTContext,
                             &xConnectInfo,
                             NULL,
                             mqttexampleCONNACK_RECV_TIMEOUT_MS,
                             &xSessionPresent );
 
     LogInfo( ( "Session present: %d", xSessionPresent ) );
-    configASSERT( xResult == MQTTSuccess );
+
+    /* Resume a session if desired. */
+    if( ( xResult == MQTTSuccess ) && !xCleanSession )
+    {
+        prvResumeSession( xSessionPresent );
+    }
+
+    return xResult;
+}
+
+/*-----------------------------------------------------------*/
+
+static void prvResumeSession( bool xSessionPresent )
+{
+    MQTTStatus_t xResult;
```

Comment:
This variable is assigned the status of the publish but it never seems to be used beyond that. Could it simply be removed or maybe queuing it in the event of failure?

Reply:
I didn't check it since I didn't want a failed publish to stop the demo, but it probably should. I think I'll change the return type for the function so if there's an error, the loop will disconnect and try the connection again.